engine/src/main/java/org/terasology/logic/players/PlayerFactory.java

```diff
@@ -93,7 +93,9 @@ public Vector3f findSpawnPositionFromLocationComponent(LocationComponent locatio
         EntityBuilder builder = entityManager.newBuilder("engine:player");
         float extraSpace = 0.5f;  // spawn a little bit above the ground
         float entityHeight = getHeightOf(builder) + extraSpace;
-        return findSpawnPos(JomlUtil.from(locationComponent.getWorldPosition()), entityHeight).get(); // TODO: Handle Optional being empty
+        return findSpawnPos(JomlUtil.from(locationComponent.getWorldPosition()), entityHeight)
```

Comment:
I think the "right" fix for this is to change the method signature to return an `Optional<Vector3f>`. Imho there is nothing _exceptional_ about a `find...` method not finding something. 
An alternative which leaves the signature as is would be to return `null` instead (another way of encoding the no spawn position was found).

I like the `Optional` better as it forces the calling site to react on the possibility of failure.

I don't think that a plain `RuntimeException` is the right thing to use here, as it is way too generic, and not really necessary. If we'd go for "exception-based programming" we'd introduce a `NoSpawnPositionFoundException` to catch further up the chain...

Reply:
Oh, this is by no means a _fix._

this is a patch to a known-broken method to produce a more recognizable failure message when it breaks.

Reply:
Right, the TODO is still there 😅 Baby step improvements ^^